From 5c1952e0f8d86ca2496abc7cc985d7f788128dd5 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Tue, 4 Dec 2012 17:42:37 +0800
Subject: [PATCH 111/304] ENGR00235659 USBDevice: Cannot recognize U-disk

Mount all udisk in /devices/platform/fsl-ehci for all host port
Make udisk mount independent on kernel usb controller and device
registing sequence

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 imx53_smd/vold.fstab    | 6 ++++--
 imx5x/init.rc           | 1 -
 imx6/arm2/vold.fstab    | 8 +++++---
 imx6/etc/init.rc        | 1 -
 imx6/evk/vold.fstab     | 8 +++++---
 imx6/sabresd/vold.fstab | 8 +++++---
 sabreauto_6q/vold.fstab | 8 +++++---
 7 files changed, 24 insertions(+), 16 deletions(-)

diff --git a/imx53_smd/vold.fstab b/imx53_smd/vold.fstab
index 8e6bd12..0f779f6 100644
--- a/imx53_smd/vold.fstab
+++ b/imx53_smd/vold.fstab
@@ -19,9 +19,11 @@ dev_mount sdcard /mnt/sdcard 1 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc0
 #for external sdcard as primary storage
 dev_mount extsd /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc2 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc2
 #used for usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.1/usb /devices/platform/fsl-ehci.0/usb
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.1/usb /devices/platform/fsl-ehci.0/usb
 #used for usb otg host
-#dev_mount udisk /mnt/sdcard/udisk auto /devices/platform/fsl-ehci.0/usb
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
+#used for all usb host
+dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
 
 
 ## Example of a dual card setup
diff --git a/imx5x/init.rc b/imx5x/init.rc
index f1e4d83..81cf243 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -34,7 +34,6 @@ loglevel 3
     export LOOP_MOUNTPOINT /mnt/obb
     export SHM_MOUNTPOINT /mnt/shm
     export EXTERNAL_STORAGE /mnt/sdcard
-    export SECONDARY_STORAGE /mnt/extsd:/mnt/udisk
     export BOOTCLASSPATH /system/framework/core.jar:/system/framework/core-junit.jar:/system/framework/bouncycastle.jar:/system/framework/ext.jar:/system/framework/framework.jar:/system/framework/android.policy.jar:/system/framework/services.jar:/system/framework/apache-xml.jar
 
     export LIGHTSENSOR_PATH /sys/class/i2c-adapter/i2c-1/1-0044
diff --git a/imx6/arm2/vold.fstab b/imx6/arm2/vold.fstab
index c353442..2344535 100644
--- a/imx6/arm2/vold.fstab
+++ b/imx6/arm2/vold.fstab
@@ -19,11 +19,13 @@
 #for external sdcard as primary storage
 #dev_mount sdcard /mnt/sdcard 4 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc0/
 dev_mount extsd  /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc2/ /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1/
-#dev_mount extsd /mnt/sdcard/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
+#dev_mount extsd /mnt/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
 #used for usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
 #used for usb otg host
-#dev_mount udisk /mnt/sdcard/udisk auto /devices/platform/fsl-ehci.0/usb
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
+#used for all usb host
+dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
 
 
 ## Example of a dual card setup
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 75e1763..ec0b73a 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -35,7 +35,6 @@ loglevel 3
     export BOOTCLASSPATH /system/framework/core.jar:/system/framework/core-junit.jar:/system/framework/bouncycastle.jar:/system/framework/ext.jar:/system/framework/framework.jar:/system/framework/android.policy.jar:/system/framework/services.jar:/system/framework/apache-xml.jar
     export SHM_MOUNTPOINT /mnt/shm
     export EXTERNAL_STORAGE /storage/sdcard0
-    export SECONDARY_STORAGE /mnt/extsd:/mnt/udisk
     export LIGHTSENSOR_PATH /sys/class/i2c-adapter/i2c-1/1-0044
     export VPU_FW_PATH /system/lib/firmware/vpu
     export CODEC_SHM_PATH /mnt/shm
diff --git a/imx6/evk/vold.fstab b/imx6/evk/vold.fstab
index 7c87706..a44fb4e 100644
--- a/imx6/evk/vold.fstab
+++ b/imx6/evk/vold.fstab
@@ -19,11 +19,13 @@
 #for external sdcard as primary storage
 dev_mount sdcard /mnt/sdcard 4 /devices/platform/sdhci-esdhc-imx.1/mmc_host/mmc0/
 dev_mount extsd  /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1/ /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc2/
-#dev_mount extsd /mnt/sdcard/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
+#dev_mount extsd /mnt/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
 #used for usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
 #used for usb otg host
-#dev_mount udisk /mnt/sdcard/udisk auto /devices/platform/fsl-ehci.0/usb
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
+#used for all usb host
+dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
 
 
 ## Example of a dual card setup
diff --git a/imx6/sabresd/vold.fstab b/imx6/sabresd/vold.fstab
index 2bd145c..db230c5 100644
--- a/imx6/sabresd/vold.fstab
+++ b/imx6/sabresd/vold.fstab
@@ -18,11 +18,13 @@
 #dev_mount sdcard /mnt/sdcard auto /devices/platform/mxsdhci.2/mmc_host/mmc0
 #for external sdcard as primary storage
 #dev_mount sdcard /mnt/sdcard auto /devices/platform/sdhci-esdhc-imx.3/mmc_host/mmc0/
-#dev_mount extsd /mnt/sdcard/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
+#dev_mount extsd /mnt/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
 #used for usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
 #used for usb otg host
-#dev_mount udisk /mnt/sdcard/udisk auto /devices/platform/fsl-ehci.0/usb
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
+#used for all usb host
+dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
 #mount SDHC4 SD card /mnt/sdcard as primary storage forMX6Q SABER_LITE RevC
 dev_mount extsd /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1
 #mount SDHC3 TF card to /mnt/extsd as external storage forMX6Q SABER_LITE RevC
diff --git a/sabreauto_6q/vold.fstab b/sabreauto_6q/vold.fstab
index ce822ca..69546b6 100644
--- a/sabreauto_6q/vold.fstab
+++ b/sabreauto_6q/vold.fstab
@@ -19,11 +19,13 @@
 #for external sdcard as primary storage
 dev_mount sdcard /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1/
 dev_mount extsd  /mnt/sdcard auto /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc0/
-#dev_mount extsd /mnt/sdcard/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
+#dev_mount extsd /mnt/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
 #used for usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
 #used for usb otg host
-#dev_mount udisk /mnt/sdcard/udisk auto /devices/platform/fsl-ehci.0/usb
+#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
+#used for all usb host
+dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
 
 
 ## Example of a dual card setup
-- 
1.8.0

